File: images/toolchain/alpine-3.24/Dockerfile.lua
# Toolchain image: Lua + LuaRocks, Alpine/musl (source builds).
ARG BASE_IMAGE=toolchain-base:alpine-3.24
FROM ${BASE_IMAGE}
COPY cache/lua-5.5.1.tar.gz /tmp/lua.tar.gz
RUN set -eux; \
    tar -xzf /tmp/lua.tar.gz -C /tmp && rm /tmp/lua.tar.gz; \
    cd /tmp/lua-5.5.1; \
    make linux MYCFLAGS=-fPIC; \
    make install INSTALL_TOP=/opt/lua; \
    cd /; rm -rf /tmp/lua-5.5.1
COPY cache/luarocks-3.13.0.tar.gz /tmp/luarocks.tar.gz
RUN set -eux; \
    tar -xzf /tmp/luarocks.tar.gz -C /tmp && rm /tmp/luarocks.tar.gz; \
    cd /tmp/luarocks-3.13.0; \
    ./configure --prefix=/opt/luarocks --with-lua=/opt/lua --lua-version=5.4; \
    make; \
    make install; \
    cd /; rm -rf /tmp/luarocks-3.13.0
ENV PATH=/opt/lua/bin:/opt/luarocks/bin:$PATH
RUN lua -v && luarocks --version

File: images/toolchain/alpine-3.24/Dockerfile.zig
# Toolchain image: Zig, Alpine/musl.
#
# Zig ships a fully static binary, so the same tarball works on musl.
ARG BASE_IMAGE=toolchain-base:alpine-3.24
FROM ${BASE_IMAGE}
COPY cache/zig-x86_64-linux-0.16.0.tar.xz /tmp/zig.tar.xz
RUN mkdir -p /opt/zig && tar -xJf /tmp/zig.tar.xz -C /opt/zig --strip-components=1 && rm /tmp/zig.tar.xz
ENV PATH=/opt/zig:$PATH
RUN zig version
